nupic/research/monitor_mixin/temporal_memory_monitor_mixin.py

```diff
@@ -23,14 +23,18 @@
 Temporal Memory mixin that enables detailed monitoring of history.
 """
 
-from collections import defaultdict
+import numpy
```

Comment:
What are external imports?
import numpy
from prettytable import PrettyTable ?

By built-in you mean from nupic, nupic.research etc.?
I will add you answer to https://github.com/numenta/nupic/wiki/Python-Style-Guide because it's not there.